infra/tf/terratest.md

```diff
@@ -0,0 +1,139 @@
+
+```go
+package test
+
+import (
+    "fmt"
+    "strings"
+    "testing"
+
+    "github.com/gruntwork-io/terratest/modules/random"
+    "github.com/gruntwork-io/terratest/modules/terraform"
+    test_structure "github.com/gruntwork-io/terratest/modules/test-structure"
+)
+
+func TestTerraformAwsEcrRepo(t *testing.T) {
+    t.Parallel()
+
+    tempTestFolder := test_structure.CopyTerraformFolderToTemp(t, "../", "examples/simple")
+
+    testName := fmt.Sprintf("terratest-%s", strings.ToLower(random.UniqueId()))
+    awsRegion := "us-west-2"
+
+    terraformOptions := &terraform.Options{
+        // The path to where our Terraform code is located
+        TerraformDir: tempTestFolder,
+
+        // Variables to pass to our Terraform code using -var options
+        Vars: map[string]interface{}{
+            "test_name": testName,
+        },
+
+        // Environment variables to set when running Terraform
+        EnvVars: map[string]string{
+            "AWS_DEFAULT_REGION": awsRegion,
+        },
+    }
+
+    defer terraform.Destroy(t, terraformOptions)
+    terraform.InitAndApply(t, terraformOptions)
+
+}
+
+```
+
+### Other Examples
+
+* [terraform-aws-alb-web-containers](https://github.com/trussworks/terraform-aws-alb-web-containers)
+* [terrraform-aws-ecs-service](https://github.com/trussworks/terraform-aws-ecs-service)
+* [terraform-aws-logs](https://github.com/trussworks/terraform-aws-logs/)
+
+## Run manually
+
+To run these tests manually against the `trussworks-ci` AWS account you'll need AWS access in our AWS organization. You'll need help from someone in #infrasec and must follow the [setup instructions](https://github.com/trussworks/legendary-waddle/blob/master/docs/how-to/setup-new-user.md#setup-new-iam-user).
+
+You'll also need to install `aws-vault` and ensure your `./aws/config` file is setup correctly.
+
+In most of our modules, there is a `makefile` that defines `test` so you'll run the following from the root of the repo you're testing:
+
+```sh
+AWS_VAULT_KEYCHAIN_NAME=login aws-vault exec trussworks-ci -- make test
+```
+
+## Configure CircleCi to run the tests automatically
+
+### Configure CircleCi Job
+
+Add a job to the `.circleci/config` file in the repository:
+
+```yaml
+terratest:
+  docker:
+    - image: *circleci_docker_primary
+  steps:
+    - checkout
+    - restore_cache:
+        keys:
+          - pre-commit-dot-cache-{{ checksum ".pre-commit-config.yaml" }}
+          - go-mod-sources-v1-{{ checksum "go.sum" }}
+    - run:
+        name: Assume role, run pre-commit and run terratest
+        command: |
+          temp_role=$(aws sts assume-role \
+                  --role-arn arn:aws:iam::313564602749:role/circleci \
+                  --role-session-name circleci)
+          export AWS_ACCESS_KEY_ID=$(echo $temp_role | jq .Credentials.AccessKeyId | xargs)
```

Comment:
```suggestion
          export AWS_ACCESS_KEY_ID=$(jq --raw-output .Credentials.AccessKeyId <<< "$temp_role")" 
```
Just a suggestion but I think `xargs` is only being used to remove surrounding quotes, but the `--raw-output` switch can do that also.

Reply:
Hm... that's cleaner for sure. Gonna make that change.